oak-run-commons/src/main/java/org/apache/jackrabbit/oak/index/indexer/document/DocumentStoreIndexerBase.java

```diff
@@ -357,26 +369,37 @@ public void reindex() throws CommitFailedException, IOException {
                     indexParallel(flatFileStores, indexer, progressReporter);
                 } else if (flatFileStores.size() == 1) {
                     FlatFileStore flatFileStore = flatFileStores.get(0);
-                    TopKSlowestPaths slowestTopKElements = new TopKSlowestPaths(TOP_SLOWEST_PATHS_TO_LOG);
-                    indexer.onIndexingStarting();
-                    long entryStart = System.nanoTime();
-                    for (NodeStateEntry entry : flatFileStore) {
-                        reportDocumentRead(entry.getPath(), progressReporter);
-                        indexer.index(entry);
-                        // Avoid calling System.nanoTime() twice per each entry, by reusing the timestamp taken at the end
-                        // of indexing an entry as the start time of the following entry. This is less accurate, because
-                        // the measured times will also include the bookkeeping at the end of indexing each entry, but
-                        // we are only interested in entries that take a significant time to index, so this extra
-                        // inaccuracy will not significantly change the results.
-                        long entryEnd = System.nanoTime();
-                        long elapsedMillis = (entryEnd - entryStart) / 1_000_000;
-                        entryStart = entryEnd;
-                        slowestTopKElements.add(entry.getPath(), elapsedMillis);
-                        if (elapsedMillis > 1000) {
-                            log.info("Indexing {} took {} ms", entry.getPath(), elapsedMillis);
+                    try (AheadOfTimeBlobDownloader aheadOfTimeBlobDownloader = getAheadOfTimeBlobDownloaderInterface(flatFileStore, indexer)) {
```

Comment:
I would prefer if we wrap the index store, instead of adding the logic in the DocumentStoreIndexerBase.

This is because the tree store doesn't require (this type of) the AheadOfTimeBlobDownloader.

We already have the IndexStore API, which can be wrapped.

Reply:
Can you elaborate? I'm not sure I understand the suggestion. 

The AOT downloader needs to receive callbacks with the position of the indexer, in order to not go to far ahead of the indexer. That is, it requires the callback this callback: `aheadOfTimeBlobDownloader.updateIndexed(entriesRead);`. 
which is done in the code below. Where could this code be moved to?

Reply:
The AheadOfTimeBlobDownloader is an interface and the only implementation requires a FlatFileStore. So another IndexStore is not supported. I would instead extend the FlatFileStore such that it can be wrapped by the AheadOfTimeBlobDownloader and then the AheadOfTimeBlobDownloader could be an IndexStore. 

That way there is no hard dependency on the AheadOfTimeBlobDownloader here. Instead, we can use the IndexStore here. We can discuss in person tomorrow.

Reply:
As discussed in person, could you suggest a refactoring of this PR to remove the dependency from the DocumentStoreIndexerBase to the AheadOfTimeBlobDownloader? I gave it a try but I did not find any obvious and simple way to do it.